lib/train/transports/cisco_ios.rb

```diff
@@ -0,0 +1,147 @@
+# encoding: utf-8
+
+require 'train/plugins'
+require 'train/transports/ssh'
+
+module Train::Transports
+  class BadEnablePassword < Train::TransportError; end
+
+  class CiscoIOS < SSH
+    name 'cisco_ios'
+
+    option :host, required: true
+    option :user, required: true
+    option :port, default: 22, required: true
+
+    option :password, required: true
+
+    # Used to elevate to enable mode (similar to `sudo su` in Linux)
+    option :enable_password
+
+    def connection
+      validate_options(@options)
+      @connection ||= Connection.new(@options)
```

Comment:
Do you actually want to validate the options every time or should this instead return an existing connection _or_ validate and connect?

Oh, I looked up `validate_options`; it returns the validated options, so you can pass it right through. =^)
```ruby
def connection
  @connection ||= Connection.new(validate_options(@options))
end
```

Reply:
I was unable to get that to work. Looks like it adds some other stuff in the return object

See: https://gist.github.com/00c569cfe7ac85ff6eb397f8d74b7a3d

I could do: 
```
@connection ||= Connection.new(validate_options(@options).options)
```
I pushed that up in the latest, commit. Let me know if that is incorrect in some way. I definitely only want to call it if `@connection` doesn't exist.